wallet/src/de/schildbach/wallet/ui/dashpay/notification/ContactViewHolder.kt

```diff
@@ -0,0 +1,128 @@
+import java.text.SimpleDateFormat
+import java.util.*
+
+open class ContactViewHolder(inflater: LayoutInflater, parent: ViewGroup) :
+        NotificationViewHolder(R.layout.notification_contact_request_received_row, inflater, parent) {
+
+    private val dateFormat = SimpleDateFormat("MMM dd, yyyy KK:mm a", Locale.getDefault())
+
+    private fun formatDate(timeStamp: Long): String {
+        return dateFormat.format(timeStamp).replace("AM", "am").replace("PM","pm")
+    }
+
+    override fun bind(notificationItem: NotificationItem, vararg args: Any) {
+        (notificationItem as NotificationItemContact).run {
+            bind(usernameSearchResult, (args[0] as Boolean), isInvitationOfEstablished, (args[1] as Boolean), (args[2] as OnContactActionClickListener))
+        }
+    }
+
+    private fun bind(usernameSearchResult: UsernameSearchResult, isNew: Boolean, isInvitationOfEstablished: Boolean,
+                     showAvatar: Boolean, onActionClickListener: OnContactActionClickListener? = null) {
+
+        val defaultAvatar = UserAvatarPlaceholderDrawable.getDrawable(itemView.context,
+                usernameSearchResult.username[0])
+
+        itemView.apply {
+            setBackgroundResource(if (isNew) R.drawable.selectable_round_corners else R.drawable.selectable_round_corners_dark)
+            date.text = formatDate(usernameSearchResult.date)
+
+            val dashPayProfile = usernameSearchResult.dashPayProfile
+            val name = if (dashPayProfile.displayName.isEmpty()) {
+                dashPayProfile.username
+            } else {
+                dashPayProfile.displayName
+            }
+
+            val displayNameResId = when (usernameSearchResult.type) {
+                UsernameSearchResult.Type.CONTACT_ESTABLISHED -> {
+                    contact_added.setImageResource(R.drawable.ic_contact_added)
+                    val sentDate = usernameSearchResult.toContactRequest!!.timestamp
+                    val receivedDate = usernameSearchResult.fromContactRequest!!.timestamp
+                    if (sentDate > receivedDate) {
+                        R.string.notifications_you_have_accepted
+                    } else {
+                        R.string.notifications_contact_has_accepted
+                    }
+                }
+                UsernameSearchResult.Type.REQUEST_RECEIVED -> {
+                    contact_added.setImageResource(R.drawable.ic_add_contact)
+                    R.string.notifications_you_received
+                }
+                UsernameSearchResult.Type.REQUEST_SENT -> {
+                    contact_added.setImageResource(R.drawable.ic_add_contact)
+                    R.string.notifications_you_sent
+                }
+            }
+
+            @Suppress("DEPRECATION")
+            val displayNameText = context.getString(displayNameResId, "<b>$name</b>")
+            displayName.text = Html.fromHtml(displayNameText)
```

Comment:
Very good! this gets the formatting to match the designs. I had no idea how make the display name bold.

Reply:
I had few ideas, but this one seems to be the best for different languages.